core/src/commitment_service.rs

```diff
@@ -97,7 +97,7 @@ impl AggregateCommitmentService {
     fn run(
         receiver: &Receiver<CommitmentAggregationData>,
         block_commitment_cache: &RwLock<BlockCommitmentCache>,
-        subscriptions: &Arc<RpcSubscriptions>,
+        rpc_subscriptions: &Option<Arc<RpcSubscriptions>>,
```

Comment:
This isn't your fault, but `&Arc<T>` is an antipattern, and typically should just use `&T` instead. How about changing this to `Option<&RpcSubscriptions>`?